sdk/core/core-lro/test/utils/fakeOperation.ts

```diff
@@ -0,0 +1,112 @@
+import { HttpOperationResponse, RequestOptionsBase } from "@azure/core-http";
+import { PollOperationState, PollOperation } from "../../src"
+import { FakeServiceClient } from "./fakeServiceClient";
+import { FakeWebResource } from "./fakeWebResource";
+
+export interface HttpPollProperties {
+  client: FakeServiceClient,
+  requestOptions?: RequestOptionsBase,
+  initialResponse?: HttpOperationResponse,
+  previousResponse?: HttpOperationResponse,
+}
+
+interface HttpPollOperation extends PollOperation<HttpPollProperties> {
+}
+
+export class FakeOperation implements HttpPollOperation {
+  public state: PollOperationState<HttpPollProperties>;
+
+  constructor(properties: HttpPollProperties) {
+    const state: PollOperationState<HttpPollProperties> = {
+      properties
+    };
+    this.state = state;
+  }
+
+  async update(): Promise<HttpPollOperation> {
+    const { client, requestOptions, initialResponse, previousResponse } = this.state.properties;
+    let response: HttpOperationResponse;
+    
+    if (!initialResponse) {
+      response = await client.sendInitialRequest(new FakeWebResource(
+        requestOptions && requestOptions.abortSignal, // abortSignal?: AbortSignalLike,
+      ));
+      this.state.properties.initialResponse = response;
+    } else
+    if (previousResponse && previousResponse.parsedBody.isFinal) {
+      response = await client.sendFinalRequest(new FakeWebResource(
+        requestOptions && requestOptions.abortSignal, // abortSignal?: AbortSignalLike,
+      ));
+    } else {
+      response = await client.sendRequest(new FakeWebResource(
+        requestOptions && requestOptions.abortSignal, // abortSignal?: AbortSignalLike,
+      ));
+    }
+    
+    this.state.properties.previousResponse = response;
```

Comment:
If we're updating state inline, I don't really see the need for this method (or cancel) to return an instance of itself. It's only if we want to be stateless and return a new HttpPollOperation with a new state object that it becomes desirable to do so, I think.